demo09_polite_random.py

```diff
@@ -0,0 +1,17 @@
+import time
+# This bot takes random moves, chosen among the legal ones for its current
+# position, but will not step on its teammate's toes.
+
+TEAM_NAME = 'PoliteRandomBots'
+
+def move(turn, game):
+    bot = game.team[turn]
+    teammate = game.team[1 - turn]
+    x0, y0 = bot.position
+    x1, y1 = teammate.position
+    move_to_teammate = (x1 - x0, y1 - y0)
+    possible_moves = bot.legal_moves[:]
```

Comment:
You could of course avoid the bug potential by comparing the `teammate.position` with `bot.get_position(one_legal_move)` and wouldn’t have to do any calculations yourself.